### PR TITLE
FOUNDATIONS: Throw Meaningful Exception (Identity)

### LINKED ISSUE #16: FOUNDATIONS: Throw Meaningful Exception (Identity)

### PR DESCRIPTION
**Pull request by JohnnySenior**
Closes #16
@cjdutoit 
@ElbekDeveloper 
@ZafarUrakov

### CODE REVIEW

**Review by ZafarUrakov — APPROVED** [review]

File: EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            string randomMessage = GetRandomMessage();
+            int randomNumber = GetRandomNumber();
+
+            CosmosException cosmosException =
+                CreateCosmosException(randomMessage, cosmosStatusCode, randomNumber);
+
+            this.cosmosErrorBrokerMock.Setup(broker =>
+                broker.GetCosmosStatusCode(cosmosException))
+                .Returns(cosmosStatusCode);
+
```
Comment (ElbekDeveloper) [inline EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs:103]: Tab

File: EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            // when . then
+            Assert.Throws<InternalServerErrorException>(() =>
+                this.efxceptionService.ThrowMeaningfulException(cosmosException));
+        }
+
+        private CosmosException CreateCosmosException(
+            string randomMessage, HttpStatusCode cosmosStatusCode, int randomNumber)
+        {
+            return new CosmosException(
+                randomMessage, cosmosStatusCode, randomNumber, randomMessage, randomNumber);
```
Comment (ElbekDeveloper) [inline EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs:173]: Indent all params and give aliases (if, needed)

File: EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                this.efxceptionService.ThrowMeaningfulException(cosmosException));
+        }
+
+        [Fact]
+        public void ShouldThrowInternalServerErrorException()
+        {
+            // given
+            HttpStatusCode cosmosStatusCode = HttpStatusCode.InternalServerError;
+            string randomMessage = GetRandomMessage();
+            int randomNumber = GetRandomNumber();
```
Comment (ElbekDeveloper) [inline EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs:155]: I don't think this matters, so basically you can remove it from here and pass these values directly inside the `CreateCosmosException` methdo

**Comment by ElbekDeveloper** [comment]
Why is your PR doesn't have
 -> FAIL
-> PASS 
Pattern? Aren't you test-driving the feature?

**Comment by ZafarUrakov** [comment]
both projects depend on shared, and when writing tests in the first` EFxception.Cosmo`s project, we wrote TDD, but when writing tests for the Identity project, there was no point in doing Fail, because everything is Pass.

**Comment by ZafarUrakov** [comment]
Please take a look at the PRs that have already been merged. @ElbekDeveloper

File: EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+            CosmosException cosmosException =
+                CreateCosmosException(randomMessage, cosmosStatusCode, randomNumber);
```
Comment (cjdutoit) [inline EFxceptions.Identity.Cosmos.Tests/EFxceptionServiceTests.cs:158]:
```suggestion
            HttpStatusCode cosmosStatusCode = HttpStatusCode.InternalServerError;

            CosmosException cosmosException =
                CreateCosmosException(cosmosStatusCode);
```

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**